File: aveo_backend/availability/urls.py
from django.urls import path
from . import views_generics

prefix = "ta2/api/"

urlpatterns = [
    path("{}teacher/".format(prefix), views_generics.CreateViewTeacher.as_view()),
    path("{}teacher/<int:pk>/".format(prefix), views_generics.DetailsViewTeacher.as_view()),
    path("{}validslot/".format(prefix), views_generics.CreateViewValidSlot.as_view()),
    path(
        "{}availableslot/".format(prefix),
        views_generics.CreateViewAvailableSlot.as_view(),
    ),
    path(
        "{}availableslot/tid/<int:teacher_id>/".format(prefix),
        views_generics.availableslot_teacher_id,
    ),
    path(
        "{}availableslot/tid/<int:teacher_id>/delete/".format(prefix),
        views_generics.availableslot_teacher_id_list,
    ),
    path(
        "{}availableslot/tid/<int:teacher_id>/date/<str:date>/".format(prefix),
        views_generics.availableslot_teacher_id_date,
    ),
    path(
        "{}availableslot/tid/<int:teacher_id>/m/<int:month>/y/<int:year>/".format(prefix),
        views_generics.availableslot_teacher_id_month_year,
    )
]

File: aveo_frontend/src/URLs.js
// APIs URLS
export const API = {
    USER_KEY: 'id',
    BASE_URL: "http://127.0.0.1:8000/",
    TEACHER_API_URL: "ta2/api/teacher/{0}/",
    MONTH_API_URL: "ta2/api/availableslot/tid/{0}/m/{1}/y/{2}/",
    SLOT_GET_API_URL: "ta2/api/availableslot/tid/{0}/date/{1}/",
    SLOT_POST_API: "ta2/api/availableslot/",
    SLOT_DELETE_API: "ta2/api/availableslot/tid/{0}/delete/",
    VALID_SLOTS_GET_API: "ta2/api/validslot/"
}

export const format = (string, arg) => {
    var formatted = string;
    for (var i = 0; i < string.length; i++) {
        var regexp = new RegExp('\\{' + i + '\\}', 'gi');
        formatted = formatted.replace(regexp, arg[i]);
    }
    return formatted;
}
